MiniApp/Classes/signature/Security/SignatureKeyStore.swift

```diff
@@ -0,0 +1,88 @@
+internal final class SignatureKeyStore {
+    private let service: String
+    private let account: String
+
+    typealias KeysDictionary = [String: String]
+
+    init(account: String, service: String = Bundle.main.bundleIdentifier!) {
+        self.service = service
+        self.account = "\(service).\(account).keys"
+    }
+
+    func key(for keyId: String) -> String? {
+        getKeys()?[keyId]
+    }
+
+    func addKey(key: String, for keyId: String) {
+        var keysDict = getKeys() ?? [:]
+
+        keysDict[keyId] = key
+        write(keys: keysDict)
+    }
+
+    func empty() {
+        write(keys: [:])
+    }
+
+    private func write(keys: KeysDictionary) {
+        guard let data = try? JSONSerialization.data(withJSONObject: keys, options: []) else {
+            return
+        }
+
+        let queryFind: [String: Any] = [
+            kSecClass as String: kSecClassGenericPassword,
+            kSecAttrService as String: service,
+            kSecAttrAccount as String: account
+        ]
+
+        let update: [String: Any] = [
+            kSecValueData as String: data
+        ]
+
+        var status = SecItemUpdate(queryFind as CFDictionary, update as CFDictionary)
+
+        if status == errSecItemNotFound {
+            let queryAdd: [String: Any] = [
+                kSecClass as String: kSecClassGenericPassword,
+                kSecAttrService as String: service,
+                kSecAttrAccount as String: account,
+                kSecValueData as String: data,
+                kSecAttrAccessible as String: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+            ]
+            status = SecItemAdd(queryAdd as CFDictionary, nil)
+        }
+
+        if status != errSecSuccess {
+            let error: String?
+            if #available(iOS 11.3, *) {
```

Comment:
I think this #if can be removed because your target version is iOS 13.0, right?

Reply:
Of course! Thanks for the heads up